src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetCacheEntry.cs

```diff
@@ -1,11 +1,75 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using System;
+using UnityEngine;
+
 namespace Microsoft.MixedReality.SpectatorView
 {
+    [Serializable]
+    internal class AssetId
+    {
+        public static AssetId Empty { get; } = new AssetId(System.Guid.Empty, -1);
+
+        public StringGuid Guid => guid;
+
+        [SerializeField]
+        private StringGuid guid;
+
+        public long FileIdentifier => fileIdentifier;
+
+        [SerializeField]
+        private long fileIdentifier;
+
+        public AssetId(StringGuid guid, long fileIdentifier)
+        {
+            this.guid = guid;
+            this.fileIdentifier = fileIdentifier;
+        }
+
+        public override bool Equals(object obj)
+        {
+            AssetId assetId = obj as AssetId;
+            if (assetId == null)
+            {
+                return false;
+            }
+
+            return this == assetId;

```

Comment:
>this == assetId; [](start = 19, length = 16)

Can you also leave a comment on why fileIdentifier isn't necessary for equality? Or would it be reasonable to have a debug assert here if there doesn't need to be a runtime check?

Reply:
Ah, chatted offline, I misread this, please ignore :)

---
In reply to: [343786783](https://github.com/microsoft/MixedReality-SpectatorView/pull/264#discussion_r343786783) [](ancestors = 343786783)

Reply:
It's used in the operator definition